File: app/src/main/java/com/example/pokeapi/MainActivity.java
package com.example.pokeapi;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.example.pokeapi.Retrofit2.ApiRest;
import com.example.pokeapi.Retrofit2.Pokemon;
import com.example.pokeapi.Retrofit2.PokemonLista;
import com.example.pokeapi.Retrofit2.Urls;
import com.example.pokeapi.Retrofit2.Utils;

import org.w3c.dom.Text;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.Url;


public class MainActivity extends AppCompatActivity {
    public static final String TAG = "Pokedex";
    private Retrofit retrofit;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        retrofit = new Retrofit.Builder()
                .baseUrl(Utils.URL_BASE)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        Button busqueda = findViewById(R.id.btn);
        TextView Height = findViewById(R.id.txtHeight);
        TextView Name = findViewById(R.id.txt);
        TextView Pokemon = findViewById(R.id.txtPoke);


        obtenerDatos();


    }

    private void obtenerDatos(){
        //ApiRest apiRest = new ApiRest();
        Urls service = retrofit.create(Urls.class);
        Call<PokemonLista> pokemonListaCall = service.getPokeLista();

        pokemonListaCall.enqueue(new Callback<PokemonLista>() {
            @Override
            public void onResponse(Call<PokemonLista> call, Response<PokemonLista> response) {
                if(response.isSuccessful()){

                    ArrayList<Pokemon> pokemonLista = response.body().getPokedex();
                    ArrayList<Pokemon> pokedey = pokemonLista;
                    for(int i = 0; i < pokedey.size(); i++){
                        Pokemon p = pokedey.get(i);
                        System.out.println("Pokemon: " + p.getName());

                    }


                }else{
                    Log.e(TAG, "Ha ocurrido un error: " + response.errorBody());
                }
            }

            @Override
            public void onFailure(Call<PokemonLista> call, Throwable t) {
                Log.e(TAG, "Error: " + t.getMessage());
            }
        });
    }
}

File: app/src/main/java/com/example/pokeapi/MainActivity.kt
package com.example.pokeapi

import android.app.Activity
import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.pokeapi.Retrofit2.ApiRest
import com.example.pokeapi.Retrofit2.PokemonLista
import com.example.pokeapi.Retrofit2.Urls
import com.example.pokeapi.Retrofit2.Utils
import com.example.pokeapi.databinding.ActivityMainBinding
import com.google.gson.JsonObject
import org.json.JSONObject
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class MainActivity : AppCompatActivity() {
    private var retrofit: Retrofit? = null
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val busqueda: Button = findViewById(R.id.btn)
        val Height: TextView = findViewById(R.id.txtHeight)
        val Vista: TextView = findViewById(R.id.txt)
        val Pokemon: EditText = findViewById(R.id.txtPoke)


        val espacio = " "


        busqueda.setOnClickListener {
            val retrofit2 = ApiRest()
            val buscando = Pokemon.text.toString()


            if (buscando.length > 50) {
                val toast = Toast.makeText(applicationContext, "La busqueda no puede exceder los 50 caracteres", Toast.LENGTH_SHORT)
                toast.show()
                Vista.text = ""
            }

            if(busqueda.text == " ") {
                val toast = Toast.makeText(applicationContext, "La busqueda no puede contener espacios", Toast.LENGTH_SHORT)
                toast.show()
                Vista.text = ""
            }

            val call: Call<JsonObject> = retrofit2.getService().getPokemonID(buscando)

            call.enqueue(object : Callback<JsonObject>{
                override fun onResponse(call: Call<JsonObject>, response: Response<JsonObject>) {
                    if (response.isSuccessful){
                        val pokemon: JsonObject? = response.body()
                        val name = pokemon!!["name"].asString


                        var datos = ("\n Nombre: " + name
                                + "\n Altura: ${pokemon!!["height"]}"
                                + "\n Peso: ${pokemon!!["weight"]}"
                                + "\n Experiencia Inicial: ${pokemon!!["base_experience"]}")
                        Vista.text = datos
                        Log.i("detalle", pokemon.toString())
                    } else{0
                        Log.e("error", "Hubo un error!")
                    }
                }

                override fun onFailure(call: Call<JsonObject>, t: Throwable) {
                    Log.e("error", t.toString())
                }
            })



        }
    }


}